### PR TITLE
Replay game

### PR DESCRIPTION
**Pull request by plopeziv**

### CODE REVIEW

File: lib/game_loop.ex [hunk 1]
```diff
@@ -0,0 +1,52 @@
+import GetUserMove
+import BestComputerSpot
+import WinningCombo
+
+defmodule GameLoop do
+
+  def _yes_no_checker(command_prompt)do
```
Comment (davetorre) [inline lib/game_loop.ex:7]: Any reason this function isn't defined inside the ElixirTicTacToe module? Looks like that's the only place where it's used.

File: lib/game_loop.ex [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    printed_board = "\n #{board[:spotOne]} | #{board[:spotTwo]} | #{board[:spotThree]} "<>
+                    "\n---+---+---\n "<>
+                    "#{board[:spotFour]} | #{board[:spotFive]} | #{board[:spotSix]} "<>
+                    "\n---+---+---\n "<>
+                    "#{board[:spotSeven]} | #{board[:spotEight]} | #{board[:spotNine]} \n"
+
+    IO.puts printed_board
+  end
+
+  def _game_loop(board, player_token) do
```
Comment (davetorre) [inline lib/game_loop.ex:24]: Public function, no underscore

File: lib/query_adapters.ex [hunk 3]
```diff
@@ -12,4 +12,21 @@ defmodule QueryAdapters do
 
     TTT.Repo.exists?(query)
   end
+
```
Comment (davetorre) [inline lib/query_adapters.ex:15]: First two functions in this file are public and shouldn't have underscores. I'd be tempted to make all of these functions public though. Seem like functions I'd want to call as a user of this module.

File: lib/token_helpers.ex [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  def _input_excluder(array, test_element) do
+    if Enum.member?(array, test_element) == false do
+      test_element
+
+    else
+      _input_excluder(array, _slug_input("Invalid Input: Please choose an available option. "))
+    end
+  end
+
+  def _create_token(message_prompt) do
```
Comment (davetorre) [inline lib/token_helpers.ex:22]: Public function, no underscore.

File: lib/elixir_tic_tac_toe.ex [hunk 5]
```diff
@@ -1,52 +1,36 @@
...
-  end
-
-  def _user_turn(board, player_token) do
-    user_board = get_move(board, player_token)
-    _print_board(user_board)
-
-    user_board
+  def _replay_game(token) do
+    if GameLoop._yes_no_checker("Do you want to play again? (Y/N): ") == "Y" do
+      clean_board = [
```
Comment (davetorre) [inline lib/elixir_tic_tac_toe.ex:25]: I see the empty board defined here and then also inside Main. Would be nice for it to only be defined once. One way would be to define a function in this module that returns a fresh board. Then both Modules could call that function.

**Comment by plopeziv** [comment]
General comment, the underscores have been replaced in the main branch. I probably pulled this after the CreateTokens pull but before the latest rebase ...... I will get that updated.

File: lib/elixir_tic_tac_toe.ex [hunk 6]
```diff
@@ -1,52 +1,36 @@
...
 
 defmodule ElixirTicTacToe do
 
   def play_TTT(board) do
 
-    _game_loop(board, return_token())
+    game_token = _player_setup()
 
-  end
+    GameLoop._game_loop(board, game_token)
```
[the remark this answers is not in the document]
Comment (plopeziv) [reply lib/elixir_tic_tac_toe.ex]: Resolved in rebase

File: lib/elixir_tic_tac_toe.ex [hunk 7]
```diff
@@ -1,52 +1,36 @@
...
-      _print_board(board)
-
-      user_board = _user_turn(board, player_token)
-
-      if is_game_over(user_board, player_token) == false do
-        IO.puts("Computer's Turn!")
-        _game_loop(_computer_turn(user_board, player_token), player_token)
-      end
+  def _player_setup() do
+    if GameLoop._yes_no_checker("Would you like to play as a new player? (Y/N): ") == "Y" do
```
[the remark this answers is not in the document]
Comment (plopeziv) [reply lib/elixir_tic_tac_toe.ex]: Resolved in rebase

File: lib/game_loop.ex (hunk 1 above)
Comment (plopeziv) [reply lib/game_loop.ex]: Resolved in rebase

File: lib/game_loop.ex (hunk 2 above)
Comment (plopeziv) [reply lib/game_loop.ex]: Resolved in rebase

File: lib/query_adapters.ex (hunk 3 above)
Comment (plopeziv) [reply lib/query_adapters.ex:15]: All functions are now public notation

File: lib/token_helpers.ex (hunk 4 above)
Comment (plopeziv) [reply lib/token_helpers.ex]: underscores removed

File: lib/elixir_tic_tac_toe.ex (hunk 5 above)
Comment (plopeziv) [reply lib/elixir_tic_tac_toe.ex]: return_clean_board has been created and injected in place of the manual declaration

**Review by davetorre — APPROVED** [review]

**Pull request merged**